actors/builtin/multisig/multisig_actor.go

```diff
@@ -279,6 +305,31 @@ func (a Actor) ChangeNumApprovalsThreshold(rt vmr.Runtime, params *ChangeNumAppr
 	return nil
 }
 
+func (ahd *ProposalHashData) Serialize() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	if err := ahd.MarshalCBOR(buf); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (a Actor) GetProposalHash(rt vmr.Runtime, txn Transaction) [32]byte {
+	hashData := ProposalHashData{
+		Requester: txn.Approved[0],
+		To:        txn.To,
+		Value:     txn.Value,
+		Method:    txn.Method,
+		Params:    txn.Params,
+	}
+
+	data, err := hashData.Serialize()
+	if err != nil {
+		rt.Abortf(exitcode.ErrIllegalState, "failed to construct multisig approval hash: %v", err)
+	}
+
+	return blake2b.Sum256(data)
```

Comment:
use `rt.Syscalls().HashBlake2b`, trying to stick with the ethereum opcode approach for these sorts of things.

Reply:
This call seems to be killing the unit tests I wrote/modded. I'm seeing the following when the unit tests are run.

```
--- FAIL: TestVesting (0.00s)
    --- FAIL: TestVesting/happy_path_full_vesting (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
    --- FAIL: TestVesting/partial_vesting_propose_to_send_half_the_actor_balance_when_the_epoch_is_hald_the_unlock_duration (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
    --- FAIL: TestVesting/fail_to_vest_more_than_locked_amount (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 19 but call succeeded
--- FAIL: TestApprove (0.00s)
    --- FAIL: TestApprove/simple_propose_and_approval (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
    --- FAIL: TestApprove/fail_approval_with_bad_proposal_hash (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 20 but call succeeded
    --- FAIL: TestApprove/fail_approve_transaction_more_than_once (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 20 but call succeeded
    --- FAIL: TestApprove/fail_approve_transaction_that_does_not_exist (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 17 but call succeeded
    --- FAIL: TestApprove/fail_to_approve_transaction_by_non-signer (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 18 but call succeeded
--- FAIL: TestCancel (0.00s)
    --- FAIL: TestCancel/simple_propose_and_cancel (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
    --- FAIL: TestCancel/fail_cancel_with_bad_proposal_hash (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 20 but call succeeded
    --- FAIL: TestCancel/signer_fails_to_cancel_transaction_from_another_signer (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 18 but call succeeded
    --- FAIL: TestCancel/fail_to_cancel_transaction_when_not_signer (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 18 but call succeeded
    --- FAIL: TestCancel/fail_to_cancel_a_transaction_that_does_not_exist (0.00s)
        mockrt.go:594: invalid runtime invocation outside of method call
        mockrt.go:507: expected abort with code 17 but call succeeded
```